### PR TITLE
2-Added GET /api/topics

### PR DESCRIPTION
**Pull request by stephenbowyer**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            });
+    });
+    test('200: should return expected field types', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                expect(typeof body.topics[0].slug).toBe('string');
+                expect(typeof body.topics[0].description).toBe('string');
+            });
```
Comment (nc-scarlett-adams) [inline __tests__/app.test.js:54]: Good tests.  Look into expect.objectContaining. It'll make your test suits looks tidy

File: app.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      response.status(400).send({msg: "Bad request"});
+}); 
+
+const server = app.listen(9090, (error) => {
+    if (error){
+        console.log(error);
+    }else{
+        console.log("Listening on 9090");
+    }
+});
```
Comment (nc-scarlett-adams) [inline app.js:21]: Please remove the code which is not related to this ticket. Put it back when it's needed later

Comment (MKD01) [inline app.js:21]: app.listen should not be defined in this file. This should either be removed or added to its own separate file.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    });    
+});
+describe('GET /api/topics', () => {
+    test('200: should return an array', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                expect(Array.isArray(body.topics)).toBe(true);
+            });
```
Comment (MKD01) [inline __tests__/app.test.js:29]: Its nice to see tests built up incrementally like this, however some of these tests are somewhat redundant. For example this first test is asserting that topics is an array, but in all of the following tests we are accessing and dealing with topics as if it is an array, which is indirectly confirming that topics is an array so you dont have to have a dedicated test for this.

File: controllers/topics.controllers.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const {selectAllTopics} = require('../models/topics.models.js');
+
+function getTopics(request, response, next){
+    selectAllTopics().then((topics) => {
+        response.status(200).send({topics});
+    });
```
Comment (nc-scarlett-adams) [inline controllers/topics.controllers.js:6]: Don't forget the .catch()

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    });
+    test('200: should return expected fields', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                expect(Object.keys(body.topics[0])).toContain('slug', 'description');
+            });
+    });
```
Comment (MKD01) [inline __tests__/app.test.js:46]: Very similar to the above, we are asserting that there should be slug and description keys within the first topic, but below we are checking that the value is of type string, which is indirectly asserting that the slug and description keys exist.

**Review by nc-scarlett-adams — COMMENTED** [review]
Good tests and solution. Just a couple of changes to make. No need to ask for another pr.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    test('200: should return expected field types', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                expect(typeof body.topics[0].slug).toBe('string');
+                expect(typeof body.topics[0].description).toBe('string');
+            });
+    });
```
Comment (MKD01) [inline __tests__/app.test.js:55]: This is also only asserting that the first topic object contains these keys with string values, but to be sure our response is correct we should ensure every topic object has these keys and values.

File: app.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+const {getTopics} = require('./controllers/topics.controllers.js');
+
+app.get('/api/topics', getTopics);
+
+app.use((err,request,response,next) => {
+    if (err.status && err.msg)
+      response.status(err.status).send({msg: err.msg});
+    else
+      response.status(400).send({msg: "Bad request"});
+});
```
Comment (MKD01) [inline app.js:13]: These should not be added for this end point as this endpoint does not use any of these error handlers. You can however add an error handler with a generic 500 status code and an appropriate message.

File: __tests__/app.test.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    return db.end().then(() => {return server.close()});
+})
+
+describe('Invalid endpoint requests', () => {
+    test('404: should return not found in response to get request to non-existent endpoint', () => {
+        return request(app)
+            .get('/api/does-not-exist')
+            .expect(404)
+    });    
+});
```
Comment (MKD01) [inline __tests__/app.test.js:21]: Although not specific to this endpoint its nice to see this added, but there should also have an expect statement here for what we want the message to be on the response object.

**Review by MKD01 — COMMENTED** [review]
Good work, but some comments have been left for you to take a look at 👍

**Pull request merged**